### ISSUE TITLE
Unexpected drop in codecoverage.

### ISSUE THREAD

**Issue by ADmad** [codecov/support#217]
The coverage for [cakephp/cakephp](https://codecov.io/gh/cakephp/cakephp/branch/master) has unexpectedly dropped from around 92% to about 41%. Can you please shed some light on what could be the cause for this. There haven't been any major changes to our test suite so must be some configuration issue or change in how codecov measures coverage.

**Comment by HavokInspiration** [comment]
If it can be of any help, the coverage for our [cakephp/migrations](https://codecov.io/github/cakephp/migrations?branch=master) plugin dropped a lot as well (from ~80% to 33%).

**Comment by ADmad** [comment]
@stevepeak Ping

**Comment by xh3b4sd** [comment]
I am experiencing a similar behaviour. In fact my experienced drop is not that big, but the reported coverage in pull requests is always higher than it then ends up in master after merging the PR. E.g. see https://github.com/xh3b4sd/anna/pull/159. Here the promised coverage is `92.28%`. [The actual commit that ended up in master](https://codecov.io/gh/xh3b4sd/anna/commit/4720cc1d71c5c31a8a1e46f0eb4239843aa42036) actually was at `89.05`. Can someone please provide some information for this? I am experiencing this for roughly a week now.

**Comment by stevepeak** [comment]
Hey guys, sorry for the delay. The best place to start would be to look at the builds and check what raw uploads are being submitted. When we get support about this topic it almost always ends up with missing coverage reports.

@ADmad did that fix above solve it?
@xh3b4sd When you review [this build](https://codecov.io/gh/xh3b4sd/anna/commit/4720cc1d71c5c31a8a1e46f0eb4239843aa42036/builds) vs [this build](https://codecov.io/gh/xh3b4sd/anna/commit/4ddf50a03faf2a5ab8cb73608fd45f19fb6afb08/builds) you we see extra reports being uploaded which I reckon contain the extra coverage data

**Comment by xh3b4sd** [comment]
Thanks for your quick reply Steve. What can I do to fix this issue? I mean I did nothing wrong or different than before. How can I prevent uploading such extra reports. I am not aware of doing this on purpose. It is only that this happened out of a sudden without me being aware of doing something wrong.

**Comment by stevepeak** [comment]
The extra upload is only a suspect in this issue, I have not reviewed the reports to confirm this, which is difficult because I'm not aware of the project and what to specifically look for.

Are you able to find specific line in a file that are you believe are causing the drop in coverage?

**Comment by xh3b4sd** [comment]
No, not really. I would need to check line by line. Lets see if I find time for that at a later point.

**Comment by ADmad** [comment]
@stevepeak No that fix didn't affect the coverage. The issue with drop in coverage stands as it is.

**Comment by ADmad** [comment]
@stevepeak Apparently something has changed in `phpdbg`. Switching to using `xdebug` to generate coverage report fixed the issue for us https://github.com/cakephp/cakephp/pull/8977

**Comment by albertodonato** [comment]
I'm experiencing a similar issue on https://github.com/albertodonato/prometheus-aioexporter.
Coverage dropped from 100% to 72.8% on a release commit that doesn't really change any code, and lines where there's supposed to be missing coverage don't really make sense (see https://codecov.io/gh/albertodonato/prometheus-aioexporter/src/9f0128430c3833295220a434abbcfb11e1c09a22/prometheus_aioexporter/script.py for instance).
Travis build passed, and codecov seems to run without issues.

**Comment by stevepeak** [comment]
Hey all - If you code base is **python** then the following may apply to your reports.

Codecov tries to resolve file paths and are joining some site-package files into your package source.
e.g., `site-packages/misc/script.py` => `prometheus_aioexporter/script.py`. 
We acknowledge this behavior is a bug that we are looking into a resolution **now**.

Some customers upload reports with dependencies included, so we cannot simply ignore them. What we will do is determine if there are package files provided along with site-packages. If both are present then site-packages will be removed automatically.

**Comment by stevepeak** [comment]
@albertodonato for example `class branch-rate="0" complexity="0" filename=".tox/coverage/lib/python3.5/site-packages/toolrack/script.py" line-rate="0.6" name="script.py">` is included in your uploads. 

Ignoring site-packages is the *best* fix for this issue: 
```
# .coveragerc
[run]
source =
  prometheus_aioexporter
```
> http://coverage.readthedocs.io/en/coverage-4.2/config.html
